python/meep.i

```diff
@@ -284,6 +285,11 @@ PyObject *py_do_harminv(PyObject *vals, double dt, double f_min, double f_max, i
   delete $1;
 }
 
+// Typemap suite for array_slice
+// TODO: add (cdouble *, int) version
+%apply (double* INPLACE_ARRAY1, int DIM1) {(double *slice, int slice_length)};
+%apply int INPLACE_ARRAY1[ANY] { int [3] };
```

Comment:
@ChrisHogan will need to add a higher-level API function (added to the `Simulation` class, probably) that does the `reshape` for you, but this is fine for now.